src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -7166,32 +7166,38 @@ HRESULT DacDbiInterfaceImpl::GetObjectFields(COR_TYPEID id, ULONG32 celt, COR_FI
     for (ULONG32 i = 0; i < cFields; ++i)
     {
         FieldDesc *pField = fieldDescIterator.Next();
-        layout[i].token = pField->GetMemberDef();
-        layout[i].offset = (ULONG32)pField->GetOffset() + (fReferenceType ? Object::GetOffsetOfFirstField() : 0);
+
+        COR_FIELD* corField = layout + i;
+        corField->token = pField->GetMemberDef();
+        corField->offset = (ULONG32)pField->GetOffset() + (fReferenceType ? Object::GetOffsetOfFirstField() : 0);
 
         TypeHandle fieldHandle = pField->LookupFieldTypeHandle();
 
         if (fieldHandle.IsNull())
         {
-            layout[i].id.token1 = 0;
-            layout[i].id.token2 = 0;
-            layout[i].fieldType = (CorElementType)0;
+            corField->id = {};
+            corField->fieldType = (CorElementType)0;
+        }
+        else if (fieldHandle.IsByRef())
+        {
+            corField->id = {};
```

Comment:
I would have expected corField->id.token2 to be set to the pointed at type of the byref basically how arrays are handled.

Reply:
Ah. This makes a lot of good sense. I see your point now. I'll update appropriately.

@davidwrighton any concerns with me ignoring the `ByReference<T>` case? Since this is contributing to the issue where we plan on removing the type, I don't think it is worth the effort but can if we think it has value.